### PR TITLE
Implement alphaem exact running in photon module

### PR DESCRIPTION
**Pull request by niclaurenti**
This PR implements the alphaem exact running in the Alpha class in the validphys.photon module.
In this way it is consistent with the running used in `EKO` for QED.
Such running also considers the mixed terms between the RGEs of alphas and alphaem (therefore they are evolved together).
Edit: Since the solution must enter in the `fiatlux` integral, and the exact solution is slower than the truncated one, the strategy is to compute $\alpha$ for some reference values in the integration range and then interpolate with `scipy` (the function is continuous since the matching is applied to $\alpha_s$ only but probably not smooth in the matching points)
Edit n. 2: interpolating with `scipy` was still too slow so I interpolated with `numpy`, that is faster

I've performed a fit with this branch and it seems to be good:
- report:
https://vp.nnpdf.science/YxTPc_wmTNursIVaGFzrzg==/
- Photon comparison with NNPDF40_ecc_qed:
https://vp.nnpdf.science/mbJ3jSqKTbCsnZkih6itLw==/#PDFscalespecs0_Basespecs1_PDFnormalize1_plot_pdfs_photon
https://vp.nnpdf.science/OfKqMEFORQSWcWcvCpTkDQ==/#PDFscalespecs0_Basespecs1_PDFnormalize1_plot_pdfs_photon
- Photon comparison with NNPDF31_ecc_qed:
https://vp.nnpdf.science/bjp3QArIT7qu91RPDUpIyg==/#PDFscalespecs0_Basespecs1_PDFnormalize1_plot_pdfs_photon

### CODE REVIEW

File: validphys2/src/validphys/photon/compute.py [hunk 1]
```diff
@@ -276,13 +310,77 @@ def alpha_em_fixed_flavor(self, q, alpha_ref, qref, nf):
...
+
+        Returns
+        -------
+        alpha_em at NLO : float
+            target value of a
+        """
+        u = 2 * np.log(q / qref)
+
+        # integration kernel
+        def rge(_t, alpha, beta_qcd_vec, beta_qcd_mix, beta_qed_vec, beta_qed_mix):
```
Comment (Zaharid) [inline validphys2/src/validphys/photon/compute.py:345]: Would be good to have some comment ere, for example with readily accessible where to check this expression.

File: validphys2/src/validphys/photon/compute.py [hunk 2]
```diff
@@ -317,30 +415,61 @@ def set_thresholds_alpha_em(self):
...
+            # compute matchings
+            match_down = compute_matching_coeffs_down(self.theory["HQ"], nf - 1)
+            alpha_s = coupl_tmp[0]
+            alphas_new = apply_match(alpha_s, self.theory['PTO'] + 1, logs[nf + 1], match_down)
+            couplings_thresh[nf] = [alphas_new, coupl_tmp[1]]
 
-        return thresh, alpha_thresh
+        return thresh, couplings_thresh
 
     def set_betas(self):
```
[the remark this answers is not in the document]
Comment (niclaurenti) [reply validphys2/src/validphys/photon/compute.py]: Hi @Zaharid , I agree with the `set_` and I changed it to `compute_`. 
However, for the main comment to make `compute_couplings_at_thresholds` a free function: the main reason why I implemented it like this is that it needs the method of the class `couplings_fixed_flavor`.
So I think that defining it outside the class is in principle doable but unnecessarily complicated.

**Comment by niclaurenti** [comment]
For whoever is going to review this: there are a lot of files changed since i merged master but the only changes i did are in
`validphys2/src/validphys/photon/compute.py`
`validphys2/src/validphys/photon/structure_functions.py`
and in the
`validphys2/src/validphys/tests/photon/`
module

**Comment by scarlehoff** [comment]
Could you rebase or cherrypick the right commits in current master? (maybe opening it as a new branch/PR if you want) otherwise we risk messing with history when we merge

**Pull request closed without merging**